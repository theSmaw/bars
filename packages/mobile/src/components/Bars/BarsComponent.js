import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import { WithBarsContainer } from 'bars-shared';

import BarComponent from '../Bar/BarComponent';

export function BarsComponent({ bars, limit }) {
  return (
    <View>
      {
        bars.map((bar, index) => (
          <BarComponent

            // Generally considered an anti-pattern to use index as a key. We could massage the data
            // that comes back to add unique identifiers for each item, but this list is simple and
            // small and it is not worth the complexity
            key={index} // eslint-disable-line react/no-array-index-key

            limit={limit}
            value={bar}
          />
        ))
      }
    </View>
  );
}

BarsComponent.propTypes = {
  bars: PropTypes.arrayOf(PropTypes.number).isRequired,
  limit: PropTypes.number.isRequired
};

export default WithBarsContainer(BarsComponent);
