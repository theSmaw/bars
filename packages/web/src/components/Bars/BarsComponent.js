import React from 'react';
import PropTypes from 'prop-types';

import { WithBarsContainer } from 'bars-shared';

import BarComponent from '../Bar/BarComponent';

export function BarsComponent({ bars, limit }) {
  return (
    <div className="bars">
      {
        bars.map((bar, index) => (
          <BarComponent

            // Generally considered an anti-pattern to use index as a key. We could massage the data
            // that comes back to add unique identifiers for each item, but this list is simple and
            // small and it is not worth the complexity
            key={index} // eslint-disable-line react/no-array-index-key

            limit={limit}
            value={bar}
          />
        ))
      }
    </div>
  );
}

BarsComponent.propTypes = {
  bars: PropTypes.arrayOf(PropTypes.number).isRequired,
  limit: PropTypes.number.isRequired
};

export default WithBarsContainer(BarsComponent);
