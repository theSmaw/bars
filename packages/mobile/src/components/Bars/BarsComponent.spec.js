import React from 'react';
import { shallow } from 'enzyme';

import { BarsComponent } from './BarsComponent';

describe('BarsComponent', () =>

  it('Should render bars', () => {
    const barsComponent = shallow(
      <BarsComponent
        bars={[1, 2, 3]}
        limit={50}
      />
    );

    expect(barsComponent).toMatchSnapshot();
  })
);
