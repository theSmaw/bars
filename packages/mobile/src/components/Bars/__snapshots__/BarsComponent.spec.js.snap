// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarsComponent Should render bars 1`] = `
<Component>
  <BarComponent
    key="0"
    limit={50}
    value={1}
  />
  <BarComponent
    key="1"
    limit={50}
    value={2}
  />
  <BarComponent
    key="2"
    limit={50}
    value={3}
  />
</Component>
`;
